### PR TITLE
A couple bug fixes for swapped assessments

### PR DESCRIPTION
**Pull request by balderdash**
Use `last_attempt_for_any` and `oldest_active_level` to load the correct assessment for students, color the progress dot in the teacher view based on the student's most recent progress, and add support to the script dsl for swapping an assessment.

### CODE REVIEW

File: dashboard/app/dsl/script_dsl.rb [hunk 1]
```diff
@@ -81,8 +81,9 @@ def level_concept_difficulty(json)
 
   string :prompt
 
-  def assessment(name)
-    level(name, {assessment: true})
+  def assessment(name, properties = {})
+    properties[:assessment] = true
+    level(name, properties)
```
Comment (Hamms) [inline dashboard/app/dsl/script_dsl.rb:86]: is this being used?

**Comment by Hamms** [comment]
LGTM!

File: dashboard/app/dsl/script_dsl.rb (hunk 1 above)
Comment (balderdash) [reply dashboard/app/dsl/script_dsl.rb:86]: `assessment <name>` is used a lot, and `assessment <name>, active: false` will be used if Baker goes ahead with the assessment swap they were asking about last week.

I guess they could just switch to `level <name>, assessment: true, active: false`, but this seems nicer.

Comment (Hamms) [reply dashboard/app/dsl/script_dsl.rb:86]: and more extensible! Looks good

**Pull request merged**